geomesa-features/geomesa-feature-avro/src/test/scala/org/locationtech/geomesa/features/avro/AvroDataFileTest.scala

```diff
@@ -141,6 +145,63 @@ class AvroDataFileTest extends Specification with AbstractAvroSimpleFeatureTest
         readFeatures.map(_.getID) must containTheSameElementsAs(features.map(_.getID))
       }
     }
+
+    "serialize byte arrays" >> {
+      val features = createComplicatedFeatures(3)
+      val tmpFile = getTmpFile
+      val dfw = new AvroDataFileWriter(new FileOutputStream(tmpFile), complexSft)
+      try {
+        features.foreach(dfw.append)
+      } finally {
+        dfw.close()
+      }
+
+      val readFeatures = getFeatures(tmpFile)
+      readFeatures.size mustEqual 3
+      val origBytes = "FOOBARBAZ+12354+\u0000\u0001\u0002\u3434".getBytes(StandardCharsets.UTF_8)
+      forall (readFeatures.map(_.getAttribute("f15").asInstanceOf[Array[Byte]])) { arr =>
+        arr mustEqual origBytes
+      }
+    }
+
+    "serialize lists and maps of byte arrays and null stuff everywhere" >> {
+      val sft = SimpleFeatureTypes.createType("bytesTest",
+        "b1:Bytes,b2:Bytes,m1:Map[String,Bytes],bl:List[Bytes],*geom:Point,dtg:Date")
```

Comment:
Also...we should probably restrict the Map keys at some point...i think that bytes actually worked as a key...but maybe we don't want to do that? not sure. I guess as long as it implements hashCode and equals properly we don't care